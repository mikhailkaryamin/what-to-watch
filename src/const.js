const AppRoute = {
  ADD_COMMENT: `/films/:id/review`,
  FILM: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  PLAYER: `/player/:id`,
  ROOT: `/`,
};

const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ButtonType = {
  ADD_REVIEW: `ADD_REVIEW`,
  PLAY: `play`,
  LIST: `list`,
};

const EscKeyName = {
  LONG: `Escape`,
  SHORT: `Esc`,
};

const FilmCardsListType = {
  MAIN: `main`,
  LIKE_THIS: `like-this`,
};

const FilmDetailedTabsType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const SpaceKeyName = {
  CODE: 32,
  TEXT: `Space`,
};

export {
  AppRoute,
  AuthStatus,
  ButtonType,
  EscKeyName,
  FilmCardsListType,
  FilmDetailedTabsType,
  SpaceKeyName,
};
