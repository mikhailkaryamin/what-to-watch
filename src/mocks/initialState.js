const initialState = {
  COMMENT: {
    comments: [{
      id: 1,
      user: {
        id: 1,
        name: `Simple name`,
      },
      rating: 9,
      text: `some text`,
      date: 1293843434,
    },
    {
      id: 2,
      user: {
        id: 2,
        name: `Simple name`,
      },
      rating: 6,
      text: `some text`,
      date: 129384343,
    },
    {
      id: 3,
      user: {
        id: 3,
        name: `Simple name`,
      },
      rating: 9,
      text: `some text`,
      date: 12938034343183,
    }],
    statusUploadComment: null,
  },
  FILMS: {
    films: [
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 1,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 3,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      },
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 2,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 2,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      },
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 3,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 3,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      }
    ],
    promoFilm: {
      backgroundImage: `img/aviator.jpg`,
      description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
      director: `Arthur Agee`,
      genre: `Comedy`,
      id: 1,
      name: `Devin Albert`,
      posterImage: `img/aviator.jpg`,
      previewImage: `img/aviator.jpg`,
      previewVideoLink: `link`,
      rating: 3,
      released: 2015,
      runTime: 100,
      scoreCount: 300,
      starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
      video: `some/link`,
    },
    statusLoad: false,
  },
  OPTIONS: {
    amountRenderFilmCard: 8,
    genre: `All genres`,
    film: null,
  },
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: 777,
      email: `mail@hotmail.com`,
      name: `Vasya`,
      avatarUrl: `/img/pic.jpg`,
    },
  }
};


export {initialState};
