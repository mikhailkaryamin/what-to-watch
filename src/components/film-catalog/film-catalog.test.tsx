import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import {FilmCardsListType} from '../../const.js';
import {initialState} from '../../mocks/initialState.js';

import FilmCatalog from './film-catalog';

const mockStore = configureStore([]);

describe(`FilmCatalog`, () => {
  const store = mockStore(initialState);

  test(`should render film catalog for main`, () => {
    const wrapper = renderer
      .create(
          <Provider store={store}>
            <Router>
              <FilmCatalog
                sign={FilmCardsListType.MAIN}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  test(`should render film catalog for like this`, () => {
    const wrapper = renderer
      .create(
          <Provider store={store}>
            <FilmCatalog
              sign={FilmCardsListType.LIKE_THIS}
            />
          </Provider>
      ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});

