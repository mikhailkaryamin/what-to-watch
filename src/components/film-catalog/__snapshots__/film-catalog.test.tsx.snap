// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCatalog should render film catalog for like this 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list like-this"
  />
</section>
`;

exports[`FilmCatalog should render film catalog for main 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Comedy
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list "
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Devin Albert"
            height="175"
            src="img/aviator.jpg"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          Devin Albert
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/films/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Devin Albert"
            height="175"
            src="img/aviator.jpg"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/2"
          onClick={[Function]}
        >
          Devin Albert
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/films/3"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Devin Albert"
            height="175"
            src="img/aviator.jpg"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/3"
          onClick={[Function]}
        >
          Devin Albert
        </a>
      </h3>
    </article>
  </div>
  
</section>
`;
